tiledb/sm/fragment/fragment_metadata.h

```diff
@@ -1453,14 +927,7 @@ class FragmentMetadata {
   /** The uri of the array the metadata belongs to. */
   URI array_uri_;
 
-  /** Set of already processed delete/update conditions for this fragment. */
-  std::unordered_set<std::string> processed_conditions_set_;
-
-  /**
-   * Ordered list of already processed delete/update conditions for this
-   * fragment.
-   */
-  std::vector<std::string> processed_conditions_;
+  OffsetsFragmentMetadata* offsets_metadata_;
```

Comment:
Can we store the actual class in a shared_ptr and cache this pointer for faster access?

Reply:
Also, as this contains the rtree as well, we need another name. I would recommend loaded_metadata.